packages/api/src/task/scheduler.ts

```diff
@@ -79,8 +82,11 @@ export class TaskScheduler {
 
     // TODO: bundle all db updates in a single transaction
     if (event.error) {
-      await this.failTask(task, event.error.message);
-      // TODO: retry task
+      if (!event.error.unretriable) {
```

Comment:
WDYT of moving the `MAX_ATTEMPTS` (actually `MAX_RETRIES`) check to here and simplifying the whole `retryTask`?
```suggestion
      if (!event.error.unretriable && (task.status.retries ?? 0) < MAX_RETRIES) {
```